core/src/main/scala/api/metadata.scala

```diff
@@ -60,8 +60,8 @@ sealed class Metadata(val attributes: Vector[Attribute]) extends ShowSelf {
 
   def to_s = if (isEmpty) "{ }" else attributes mkString ("{\n  ", "\n  ", "\n}")
 }
-object Metadata extends Metadata(Vector()) {
-  def apply(attributes: Attribute*): Metadata = attributes.foldLeft(this: Metadata)(_ set _)
+object Metadata {
+  def apply(attributes: Attribute*): Metadata = attributes.foldLeft(new Metadata(Vector.empty))(_ set _)
```

Comment:
In this branch I took a different approach: https://github.com/suffuse/sfs/blob/metadata-only-experiment/core/src/main/scala/api/metadata.scala#L145